### ISSUE TITLE
Feature: Allow disabling of @lexical/link href sanitization

### ISSUE THREAD

**Issue by Slapbox** [facebook/lexical#4353]
There are valid use cases for less common protocols and they should be easy to enable for use cases where that is necessary and not foolish from a security standpoint. I think ideally it would accept a custom validation function.

**Comment by acywatson** [comment]
> There are valid use cases for less common protocols and they should be easy to enable for use cases where that is necessary and not foolish from a security standpoint. I think ideally it would accept a custom validation function.

Yes, true, I did think about this and figured we would have to consider improving the configurability API at some point in the future. However, it's totally possible now in much the same way that you would customize any other node behavior. Simply [override the node](https://lexical.dev/docs/concepts/node-replacement) with your own node and implement the sanitizeUrl function however you want. If you want it can be:

```
sanitizeUrl(url: string): string  {
  return url;
}
```

**Comment by addlistener** [comment]
currently if I do this in `sanitizeUrl`

```

export function sanitizeUrl(url: string): string {
  return url; // DIRECTLY return! 

  try {
    const parsedUrl = new URL(url);
    // eslint-disable-next-line no-script-url
    if (!SUPPORTED_URL_PROTOCOLS.has(parsedUrl.protocol)) {
      return 'about:blank';
    }
  } catch {
    return url;
  }
  return url;
}
```

The result will still be 'https://' even if I entered sth like "#footer"
